web/src/server_events_dispatch.js

```diff
@@ -662,15 +659,12 @@ export function dispatch_normal_event(event) {
                         settings_streams.update_default_streams_table();
                         stream_data.remove_default_stream(stream_id);
                         if (realm.realm_new_stream_announcements_stream_id === stream_id) {
-                            realm.realm_new_stream_announcements_stream_id = -1;
```

Comment:
This was not needed, since for cases when stream used for setting cannot be accessed by a user, we show "Cannot view channel" and the setting value is still the original stream ID value. "-1" value is for cases when that setting is disabled, which we do when archiving.

Reply:
These whole blocks might not need to exist, since the server already sends events to update those fields directly.

Reply:
This is probably worth checking if we can delete these blocks in a follow-up PR.

Reply:
Rest of the block is needed here. This is for user losing access to a stream and the setting does not change, it is just that  the user who cannot access the stream will see "Cannot view channel" in the dropdown.